From 11837756078ffff8f94bcaeeef4bdc07b0a160ed Mon Sep 17 00:00:00 2001
From: Minu <minu.mathai@intel.com>
Date: Tue, 1 Mar 2016 14:25:22 +0000
Subject: [PATCH 2/5] FROM_UPSTREAM [VPG]: drm/i915: Free wa_batchbuffer when
 freeing error state

wa_batchbuffer is part of some error states. Make sure it
is freed.

Change-Id: Idd85ad68eece65c20dc6e6d30f65d4481db31fad
Signed-off-by: Mika Kuoppala <mika.kuoppala@intel.com>
Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-22278
Upstream-repo: http://cgit.freedesktop.org/drm-intel/
Upstream-commit: b3da4a627eb710254e09c6b8dbe94a4aa35382d2
Signed-off-by: Minu Mathai <minu.mathai@intel.com>
Reviewed-on: https://android.intel.com:443/476899
---
 drivers/gpu/drm/i915/i915_gpu_error.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/i915/i915_gpu_error.c b/drivers/gpu/drm/i915/i915_gpu_error.c
index ff5ff42..7684c0c 100644
--- a/drivers/gpu/drm/i915/i915_gpu_error.c
+++ b/drivers/gpu/drm/i915/i915_gpu_error.c
@@ -539,6 +539,7 @@ static void i915_error_state_free(struct kref *error_ref)
 
 	for (i = 0; i < ARRAY_SIZE(error->ring); i++) {
 		i915_error_object_free(error->ring[i].batchbuffer);
+		i915_error_object_free(error->ring[i].wa_batchbuffer);
 		i915_error_object_free(error->ring[i].ringbuffer);
 		i915_error_object_free(error->ring[i].hws_page);
 		i915_error_object_free(error->ring[i].ctx);
-- 
1.9.1

